src/uu/sed/src/processor.rs

```diff
@@ -175,8 +195,20 @@ fn substitute(
 
     let text = pattern.as_str()?;
 
-    for caps in sub.regex.captures_iter(text) {
+    let regex = re_or_saved_re(&sub.regex, context)?;
+
+    for caps in regex.captures_iter(text) {
         count += 1;
+        let caps = match caps {
+            Ok(c) => c,
+            Err(e) => {
+                return Err(USimpleError::new(
+                    2,
+                    format!("regular expression capture retrieval error: {}", e),
+                ));
+            }
+        };
```

Comment:
as you wish but you could write:

```suggestion
     let caps = caps
    .map_err(|e| USimpleError::new(
        2,
        format!("regular expression capture retrieval error: {}", e),
    ))?;

```

Reply:
Thanks, fixed!